### ISSUE TITLE
loclist not auto-closing via manual triggered linting when LSP servers are defined

### ISSUE THREAD

**Issue by bluz71** [w0rp/ale#2328]
Hello ALE team,

I favour manual triggered linting as defined by the following:

```viml
let g:ale_lint_on_enter = 0
let g:ale_lint_on_filetype_changed = 0
let g:ale_lint_on_insert_leave = 0
let g:ale_lint_on_save = 0
let g:ale_lint_on_text_changed = 'never'
```

I then like to enable `let g:ale_open_list = 1` to display the loclist when there are lint errors. All works well if I set one linter per filetype, such as `eslint` or `rubocop`, the loclist will display when there are errors and disappear when there are no errors upon a manual triggered `:ALELint`.

However, when I specify LSP servers in addition to the linters in `g:ale_linters` the loclist will remain open even when all errors have been corrected. An example configuration:

```viml
    let g:ale_linters = {
    \  'javascript': ['eslint', 'tsserver'],
    \  'ruby':       ['rubocop', 'solargraph']
    \}
```

Steps to produce the issue:

- Have a JavaScript or Ruby file with errors
- Run `:ALELint`, loclist will open with errors (correctly)
- Fix the errors and run `:ALELint` again, signs column will clear (correctly), but loclist will remain open (incorrectly).

Now remove the `tsserver` and `solargraph` entries from `g:ale_linters` and try again. Notice that the loclist will correctly close once all errors have been corrected. This is the wanted behaviour even when multiple linters are provided.

Thank you.

**Comment by bluz71** [comment]
Note, on the Ruby side, the LSP server, **solargraph**, needs to be active for the above issue to manifest, which means something like `:ALEComplete` needs to have been triggered **before** doing the steps listed above.

On the JavaScript side no side hoop-jumping need occur.

**Comment by w0rp** [comment]
I can't repeat the bug myself.

**Comment by bluz71** [comment]
Hopefully these steps produce the issue:

- Please make sure **eslint** and **tsserver** are installed on the host

- Clone this repo: https://github.com/bluz71/platters_app

- ALE configuration in *vimrc*:

```viml
Plug 'w0rp/ale'
    let g:ale_linters = {
    \  'javascript': ['eslint', 'tsserver']
    \}
    let g:ale_lint_on_enter            = 0
    let g:ale_lint_on_filetype_changed = 0
    let g:ale_lint_on_insert_leave     = 0
    let g:ale_lint_on_save             = 0
    let g:ale_lint_on_text_changed     = 'never'
    let g:ale_linters_explicit         = 1
    let g:ale_open_list                = 1
```

- Copy my [eslintrc.yml](https://github.com/bluz71/dotfiles/blob/master/eslintrc.yml) into the base directory of the checked out repo (prepend a dot to the name so you end up with `.eslintrc.yml`).

- Open the file `src/components/AlbumShowPage.js`

- At the end of the first line delete the semi-colon

- Run `ALELint`

- One error should be displayed (correctly) complaining about missing semi-colon

- Add the semi-colon back and run `ALELint` again. The error will disappear (correctly), but the loclist window will not close.

If you remove the `tsserver` entry and rerun those same steps the loclist will correctly close if there are zero errors.
 
I believe the loclist should always be closed once there are zero errors.

Thank you for a genuinely useful plugin.

**Comment by dfalling** [comment]
This happens sporadically for me as well- I'll clean up all my errors but the loclist usually stays open.

**Comment by w0rp** [comment]
@bluz71 I can't repeat the bug with that. This may have been fixed by the patch for #2415. The loclist doesn't close until all results have been collected, and `tsserver` wasn't being marked as inactive. `git pull` and try again.

**Comment by w0rp** [comment]
Duplicate of #2415. Continue discussion there.

**Comment by bluz71** [comment]
@w0rp,

Yes, this issue is now fixed (likely commit [fcc2c3b](https://github.com/w0rp/ale/commit/fcc2c3ba71afa2a7965f3c1e9ec8c03381178180)).

Many thanks.
